concourse/model/traits/component_descriptor.py

```diff
@@ -139,6 +147,9 @@ def callback_env(self) -> dict:
     def validation_policies(self):
         return ()
 
+    def upload_component_descriptor(self):
+        return self.raw['upload_component_descriptor']
+
     def ctx_repository(self) -> model.ctx_repository.CtxRepositoryCfg:
```

Comment:
how about adding a second attribute `snapshot_ctx_repository` (note: use cfg-element-name, as for this getter). if set, the following semantics applies:

1. component-descriptor-step _always_ uploads to configured (snapshot-)ctx-repository
2. release-trait should do a "promote"; second step can be omitted as a shortcut to get this change done a bit quicker